flake.nix

```diff
@@ -383,6 +383,7 @@
 
             doInstallCheck = finalAttrs.doCheck;
             installCheckFlags = "sysconfdir=$(out)/etc";
+            installCheckTarget = "installcheck"; # work around buggy detection in stdenv
```

Comment:
Good question, FWIW I had trouble reproducing it locally.

Reply:
Not answering your question, but I suspect from past discussions that it's from [this `+`](https://github.com/NixOS/nix/blob/0159dfad3f48105ecc971d93a562aec36d15ad4a/mk/libraries.mk#L95) which breaks `make -n` (because that's magic for “recursive make” and make executes these rules regardless of `-n`)

Reply:
Yes, it's indeed the `+` that forces `make -n` to execute the command anyway. I don't understand why it's there in the first place...

Reply:
Oh huh. Good eye. That's very spooky magic...

Reply:
> I don't understand why it's there in the first place...

Glad I am not the only one! :)

Reply:
Sounds like some future work may be to remove the `+` and see what breaks, since it's unknown and can cause subtle breakage itself :eyes:

Reply:
Quick blame shows it was introduced (and discussed) here: [`2799fe4` (#5813)](https://github.com/NixOS/nix/pull/5813/commits/2799fe4cdbe77e017544f2fe61ae9baef650dbe6#r1099608721)

Reply:
Or we just https://github.com/NixOS/rfcs/pull/132 :grinning:

Reply:
> https://github.com/NixOS/nix/pull/5813/commits/2799fe4cdbe77e017544f2fe61ae9baef650dbe6#r1099608721

Oh that's very interesting. And arguably Ninja and the LTO stuff should use same jobserver things too so my Meson joke is not the full story.

I wonder if the variable can just expand to put in the `+` or that doesn't work.

Reply:
> I wonder if the variable can just expand to put in the `+` or that doesn't work.

Google tells me that we might be able to [hack our way out](https://github.com/JGRennison/retcon/blob/a40cc392ddd1a81335eca3da4a98c97da582185a/makefile#L71-L74). But that's quite nasty.